### DISCUSSION TITLE
Why might I not be able to achieve smooth shading

### DISCUSSION THREAD

**Discussion by Zardukah** [pyvista/pyvista#1776] [Q&A] [answered]
I have a model that I'm trying to achieve smooth shading on, however it doesn't appear to be having a noticeable effect. I've done smooth shading on the sphere and it works fine. What might I consider trying to achieve the effect I'm looking for?
```
import pyvista

pyvista.set_plot_theme("document")

a = pyvista.PolyData("models\monkey.obj")
b = pyvista.PolyData("models\monkey.obj")
c = pyvista.Sphere()


p = pyvista.Plotter(shape=(1,3))
p.subplot(0,0)
p.add_mesh(a)
p.subplot(0,1)
p.add_mesh(b, smooth_shading = True)
p.subplot(0,2)
p.add_mesh(c, smooth_shading = True)

p.show()

```
![image](https://user-images.githubusercontent.com/78458206/140000323-76b96a38-b24c-49e5-9b49-81eac84a0b33.png)

It seems to work with a .stl file, why is this?

![image](https://user-images.githubusercontent.com/78458206/140001069-be729352-0bef-4520-a4b1-e19e3dd61cd6.png)

I think it has something to do with the normals, but I don't understand enough to figure out how.

.stl
![image](https://user-images.githubusercontent.com/78458206/140001434-40e179aa-6310-4638-8954-aaf185cec74d.png)

.obj
![image](https://user-images.githubusercontent.com/78458206/140001422-ee81030c-4b26-4136-8b95-93413b399eb5.png)

**Comment by akaszynski** [comment, marked as answer]
This is most likely because the individual facets from the obj file are separated.  I bet if you [clean](https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyData.clean.html) the polydata after loading the "obj", you'll get similar results as the "stl".

**Reply by Zardukah** [reply]
Thank you, this seemed to work! Do you know if it would be possible to do smooth shading with certain edges still hard? 
Example: (Blender auto smooth, which I'm hoping to replicate here)
![image](https://user-images.githubusercontent.com/78458206/140043705-bbd25fbe-833a-4cf0-8ff3-30e48d3b6eb5.png)
Perhaps some way to split the edges by angle after smooth shading occurs?

**Reply by akaszynski** [reply]
Actually, yes, but not quite yet in this release.  It's a feature we're adding in:

https://github.com/pyvista/pyvista/pull/1685

**Reply by Zardukah** [reply]
> Actually, yes, but not quite yet in this release. It's a feature we're adding in:
> 
> #1685

Great work you guys do here, I excitedly await that feature! Thanks for the help

**Reply by akaszynski** [reply]
Will be out in the next release by the end of this month.
